src/components/screens/group-dialog-screen.tsx

```diff
@@ -0,0 +1,394 @@
+      if (!matchedGroup?.id) return;
+
+      setLoading(true);
+      try {
+        await Api.getGroupJoinRequestsApi(keycloak.token).createGroupJoinRequest({
+          groupId: matchedGroup.id,
+          groupJoinRequest: {
+            email: keycloak.tokenParsed.email,
+            groupId: matchedGroup.id
+          }
+        });
+
+        setDialogStatus(GroupDialogStatus.REQUEST_SENT);
+        setLoading(true);
+      } catch (error) {
+        errorContext.setError(strings.errorHandling.groupDialogsScreen.createGroupRequest);
+      }
+    } else {
+      errorContext.setError("Joining mutliple groups not yet supported");
+    }
+  };
+
+  /**
+   * Renders the welcome dialog
+   */
+  const renderWelcomeDialog = () => (
+    <GenericDialog
+      error={ false }
+      open={ true }
+      onClose={ () => {} }
+      onCancel={ () => setDialogStatus(GroupDialogStatus.JOIN) }
+      onConfirm={ () => setDialogStatus(GroupDialogStatus.CREATE) }
+      title={ strings.groupDialogsScreen.welcomeDialog.title }
+      positiveButtonText={ strings.groupDialogsScreen.createNewGroup }
+      cancelButtonText={ strings.groupDialogsScreen.welcomeDialog.joinGroup }
+      hideClose={ true }
+    >
+      <Typography>
+        { strings.groupDialogsScreen.welcomeDialog.content }
+      </Typography>
+    </GenericDialog>
+  );
+
+  /**
+   * Renders the create dialog
+   */
+  const renderCreateDialog = () => (
+    <GenericDialog
+      error={ false }
+      open={ true }
+      onClose={ () => {} }
+      onCancel={ () => setDialogStatus(GroupDialogStatus.WELCOME) }
+      onConfirm={ createGroup }
+      title={ strings.groupDialogsScreen.createDialog.title }
+      positiveButtonText={ strings.groupDialogsScreen.createNewGroup }
+      cancelButtonText={ strings.groupDialogsScreen.cancel }
+      disabled={ !newGroupName }
+      hideClose={ true }
+    >
+      <TextField
+        fullWidth
+        name="newGroupName"
+        value={ newGroupName }
+        placeholder={ strings.groupDialogsScreen.createDialog.placeholder }
+        onChange={ onNewGroupNameChange }
+        error={ groupNameError }
+        helperText={ groupNameError && strings.groupDialogsScreen.createDialog.helperText }
+      />
+    </GenericDialog>
+  );
+
+  /**
+   * Renders the create done dialog
+   */
+  const renderCreateDoneDialog = () => (
+    <GenericDialog
+      open={ true }
+      onClose={ () => {} }
+      onConfirm={ () => navigate(`/groups/${usersGroups[0].id}`)}
+      title={ strings.groupDialogsScreen.createDoneDialog.title }
+      positiveButtonText={ strings.groupDialogsScreen.createDoneDialog.goToGroupManagement }
+      hideClose={ true }
+    >
+      <Typography>
+        { strings.groupDialogsScreen.createDoneDialog.content }
+      </Typography>
+    </GenericDialog>
+  );
+
+  /**
+   * Render the join group dialog
+   */
+  const renderJoinDialog = () => {
+    const isChecked = joinGroupCheckedItems.some(checked => checked);
```

Comment:
I believe this is the correct comparison?
```suggestion
    const isChecked = joinGroupCheckedItems.some(item => item.checked);
```